pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -316,6 +337,19 @@ public void run() {
           if (_receivedStop) {
             break;
           }
+
+          if (_state == State.INITIAL_CONSUMING) {
+            initialConsumptionEnd = now();
+            _serverMetrics.setValueOfTableGauge(_metricKeyName,
+                ServerGauge.LAST_REALTIME_SEGMENT_INITIAL_CONSUMPTION_DURATION_SECONDS,
+                TimeUnit.MILLISECONDS.toSeconds(initialConsumptionEnd - _startTimeMs));
+          } else if (_state == State.CATCHING_UP) {
+            long catchUpTimeMillis = now() - initialCatchUpStart;
```

Comment:
Maybe we don't want to consider the catchup time at all? Not sure what we are gaining by it, since we have the segment completion time minus initial consumption end time as well. That metric should give us the time taken to complete a segment

Alternatively, we should add up times for all the catch up we do, and post it when exiting the FSM